File: Library.WebCore/src/app/models/bookViewModel/getBookViewItem.ts
import { GetPublicationHouseViewItem } from './../publicationHouseViewModel/getPublicationHouseViewItem';
import { GetAuthorViewItem } from './../authorViewModel/getAuthorViewItem';

export class GetBookViewItem {
  public id: number;
  public name: string;
  public authors: GetAuthorViewItem[];
  public dateOfPublishing: Date;
  public publicationHouses: GetPublicationHouseViewItem[];
}

File: Library.ViewModels/BookViewModel/GetBookViewItem.cs
using Library.ViewModels.AuthorViewModel;
using Library.ViewModels.PublicationHouseViewModel;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Library.ViewModels.BookViewModel
{
    public class GetBookViewItem
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public DateTime DateOfPublishing { get; set; }
        [Required]
        public ICollection<GetAuthorViewItem> Authors { get; set; }
        [Required]
        public ICollection<GetPublicationHouseViewItem> PublicationHouses { get; set; }
    }
}